### DISCUSSION TITLE
how to use npm modules in Cocos Creator 3D?

### DISCUSSION THREAD

**Discussion by Sorrowful-free** [cocos/cocos-engine#7481] [General] [answered]
What about using npm-packages in new version of CocosCreator [3d-v1.2](https://github.com/cocos-creator/engine/tree/3d-v1.2).

I answer because, structure of project changed, and now `package.json` use for Creator, in previous versions(2.4 for example) for Creator used `project.json`. 

I hope when you merge 2d and 3d creator you pay attention for this point, thank you so much!

**Comment by Sorrowful-free** [comment]
I found in your examples repository [`protobuf` example](https://github.com/cocos-creator/example-3d/tree/v1.2/protobuf), but you copy `protobuf` in `thirdy` but it not comfortable, just because packages can be changes, thank you!

**Comment by shrinktofit** [comment, marked as answer]
I'm afraid it won't be included in 3.0 preview. Maybe available in rc-1 or later.
